pipelines/batch/src/main/java/com/google/fhir/analytics/FetchSearchPageFn.java

```diff
@@ -219,28 +221,37 @@ public void setup() throws SQLException, ProfileException {
   }
 
   /**
+   * Note this is a hacky solution to address a DataflowRunner specific issue where @Teardown
+   * method is not guaranteed to be called. Closing/flushing Parquet files in @FinishBundle is
+   * not a good idea in general because it makes the size of those files a function of how Beam
+   * divides the work into Bundles. That's why we only do this on DataflowRunner which tend to
+   * have very large Bundles.
+   *
    * This should be overridden by all subclasses because the `context` type is not fully specified
    * at this parent class (because of the T type argument). All subclass implementations should call
    * `super.finishBundle` though. TODO: implement a way to enforce this at compile time; this is
    * currently caught at run time.
    */
   @FinishBundle
   public void finishBundle(FinishBundleContext context) {
-    try {
-      if (parquetUtil != null) {
-        parquetUtil.flushAll();
+    if (DATAFLOW_RUNNER.equals(context.getPipelineOptions().getRunner().getSimpleName())) {
```

Comment:
I was talking about this [bundleFinalizer](https://beam.apache.org/releases/javadoc/2.33.0/index.html?org/apache/beam/sdk/transforms/DoFn.BundleFinalizer.html) function on the other day. But, these seems to be tightly coupled with finish bundle, so I am assuming that this might be called for every one message that is committed for DirectRunner.

Reply:
Right; and thanks.